garage/tf/optimizers/lbfgs_optimizer.py

```diff
@@ -27,29 +25,32 @@ def update_opt(self,
                    target,
                    inputs,
                    extra_inputs=None,
-                   name="update_opt",
+                   name=None,
                    *args,
                    **kwargs):
         """
         :param loss: Symbolic expression for the loss function.
         :param target: A parameterized object to optimize over. It should
          implement methods of the
         :class:`garage.core.paramerized.Parameterized` class.
-        :param leq_constraint: A constraint provided as a tuple (f, epsilon), of
-         the form f(*inputs) <= epsilon.
+        :param leq_constraint: A constraint provided as a tuple (f, epsilon),
+         of the form f(*inputs) <= epsilon.
         :param inputs: A list of symbolic variables as inputs
         :return: No return value.
         """
-        with enclosing_scope(self._name, name):
-            self._target = target
+        self._target = target
+        params = target.get_params(trainable=True)
+        with tf.name_scope(name, "LbfgsOptimizer",
```

Comment:
I'm a little confused about the naming here. Why call it "LbfgsOptimizer" here and not in `__init__`? Will `update_opt()` only be called once?

Reply:
Nvm I think you answered it below, but I was wondering if you could still clarify for me as I'm still a little confused